src/scheduler/scheduler.ts

```diff
@@ -0,0 +1,181 @@
+import { WorkflowContext } from "..";
+import { DBOSExecutor } from "../dbos-executor";
+import { MethodRegistration, registerAndWrapFunction } from "../decorators";
+import { TimeMatcher } from "./crontab";
+import { Workflow } from "../workflow";
+
+////
+// Configuration
+////
+
+export enum SchedulerConcurrencyMode {
+    ExactlyOncePerInterval = 'ExactlyOncePerInterval',
+}
+
+export class SchedulerConfig {
+    crontab: string = '* * * * *'; // Every minute
+    mode ?: SchedulerConcurrencyMode = SchedulerConcurrencyMode.ExactlyOncePerInterval;
+}
+
+////
+// Method Decorator
+////
+
+// Scheduled Time. Actual Time, number running globally, number running locally
+export type ScheduledArgs = [Date, Date, number, number]
+export type ScheduledArgsAsSerialized = [string, string, number, number]; // You would think this is a Date but json doesn't have dates, so it is a string.
```

Comment:
So the problem is when we run `JSON.stringify` the `Date` will be converted into `string`, and when we try to recover workflow or replay a workflow there's no way to deserialize it back to `Date`?

Does this mean the way we record workflow input/output would also be all wrong for `Date` type? If so, we need to properly document this behavior and don't allow `Date` in input / output. The issue is if the scheduled workflow gets interrupted during execution, when we recover pending workflows, we need to read recorded input back through `JSON.parse()`. If we parse back a `string` not `Date`, then workflow code that relies on it to be a `Date` can go wrong.

Reply:
The `Date` type is handled in the wrapper function so there is no reason to take it away.  It just isn't a `Date` right here.

Reply:
I think when we record workflow inputs, we use JSON.stringify() to serialize inputs and store in the database. So if we recover the workflow from recorded input it would get string not Date. Is there any other places where the Date type is converted to string?

I understand the validator converts string back to Date, but maybe a better solution is to implement special replacer/reviver for JSON (can be a separate PR)..

Reply:
We can do more, but I think the validator is a good place to do it... in case it comes from HTTP message.